src/clj/game/cards/ice.clj

```diff
@@ -400,12 +400,18 @@
    {:subroutines [end-the-run]}
 
    "Blockchain"
-   (letfn [(sub-count [corp] (int (/ (count (filter #(and (is-type? % "Operation") (has-subtype? % "Transaction"))
+   (letfn [(sub-count [corp] (int (/ (count (filter #(and (is-type? % "Operation")
+                                                          (has-subtype? % "Transaction")
+                                                          (:seen %))
                                                     (:discard corp)))
                                      2)))]
-     {:abilities [{:label "Gain subroutines"
-                   :msg (msg (let [c (sub-count corp)]
-                               (str "gain " c (pluralize " subroutine" c))))}]
+     {:implementation "Number of subs is manual"
+      :abilities [{:label "Gain subroutines"
+                   :effect (req (let [c (sub-count corp)]
+                                  (update! state :corp
+                                           (assoc-in card [:special :extra-subs] (pos? c)))
```

Comment:
Will calling `pos?` here return a boolean? If so, is that what we want?

Reply:
Yes, should return a boolean. `:extra-subs` is just an indicator of extra subs, not a count. Extra subs just shows up as `+` icon on the card. Modified ice strength shows up as a number icon.